workloads/wmt/wmt_jax/train.py

```diff
@@ -0,0 +1,381 @@
+                     (str(logits.shape), str(targets.shape)))
+  loss = jnp.equal(jnp.argmax(logits, axis=-1), targets)
+  normalizing_factor = np.prod(logits.shape[:-1])
+  if weights is not None:
+    loss = loss * weights
+    normalizing_factor = weights.sum()
+
+  return loss.sum(), normalizing_factor
+
+
+def compute_metrics(logits, labels, weights, label_smoothing=0.0):
+  """Compute summary metrics."""
+  loss, weight_sum = compute_weighted_cross_entropy(logits, labels, weights,
+                                                    label_smoothing)
+  acc, _ = compute_weighted_accuracy(logits, labels, weights)
+  metrics = {
+      "loss": loss,
+      "accuracy": acc,
+      "denominator": weight_sum,
+  }
+  metrics = jax.lax.psum(metrics, axis_name="batch")
+  return metrics
+
+
+# Primary training / eval / decode step functions.
+# -----------------------------------------------------------------------------
+
+
+def train_step(optimizer,
+               batch,
+               config,
+               learning_rate_fn,
+               label_smoothing=0.0,
+               dropout_rng=None):
+  """Perform a single training step."""
+  # X_position and X_segmentation are needed only when using "packed examples"
+  # where multiple sequences are packed into the same example with this
+  # metadata.
+  # if such features are not present they are ignored and the example is treated
+  # like a normal, unpacked sequence example.
+  train_keys = [
+      "inputs", "targets", "inputs_position", "targets_position",
+      "inputs_segmentation", "targets_segmentation"
+  ]
+  (inputs, targets, inputs_positions, targets_positions, inputs_segmentation,
+   targets_segmentation) = [batch.get(k, None) for k in train_keys]
+
+  weights = jnp.where(targets > 0, 1, 0).astype(jnp.float32)
+
+  dropout_rng = jax.random.fold_in(dropout_rng, optimizer.state.step)
+
+  def loss_fn(params):
+    """loss function used for training."""
+    logits = models.Transformer(config).apply(
+        {"params": params},
+        inputs,
+        targets,
+        inputs_positions=inputs_positions,
+        targets_positions=targets_positions,
+        inputs_segmentation=inputs_segmentation,
+        targets_segmentation=targets_segmentation,
+        rngs={"dropout": dropout_rng})
+
+    loss, weight_sum = compute_weighted_cross_entropy(logits, targets, weights,
+                                                      label_smoothing)
+    mean_loss = loss / weight_sum
+    return mean_loss, logits
+
+  step = optimizer.state.step
+  lr = learning_rate_fn(step)
+  grad_fn = jax.value_and_grad(loss_fn, has_aux=True)
+  (_, logits), grad = grad_fn(optimizer.target)
+  grad = jax.lax.pmean(grad, "batch")
+  new_optimizer = optimizer.apply_gradient(grad, learning_rate=lr)
+  metrics = compute_metrics(logits, targets, weights)
+  metrics["learning_rate"] = lr
+
+  return new_optimizer, metrics
+
+
+def eval_step(params, batch, config, label_smoothing=0.0):
```

Comment:
lets remove `label_smoothing` from here and `compute_metrics`, because a non-default value is never actually passed in